sale_report_margin/reports/sale_report.py

```diff
@@ -0,0 +1,15 @@
+# Copyright 2018 Tecnativa - Sergio Teruel
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import fields, models
+
+
+class SaleReport(models.Model):
+    _inherit = "sale.report"
+
+    purchase_price = fields.Float(readonly=True)
+
+    def _select_additional_fields(self):
+        res = super()._select_additional_fields()
+        res["purchase_price"] = "l.purchase_price"
```

Comment:
You have to overwrite `_select_sale` instead and do `SUM(l.purchase_price)` for having this correctly and not grouping by.

Reply:
Well, maybe this can serve:

```suggestion
        res["purchase_price"] = "SUM(l.purchase_price)"
```